x-pack/plugins/gis/public/shared/layers/vector_layer.js

```diff
@@ -199,29 +201,36 @@ export class VectorLayer extends ALayer {
 
     let updateDueToTime = false;
     if (timeAware) {
-      updateDueToTime = !_.isEqual(meta.timeFilters, filters.timeFilters);
+      updateDueToTime = !_.isEqual(meta.timeFilters, searchFilters.timeFilters);
     }
 
     let updateDueToRefreshTimer = false;
-    if (refreshTimerAware && filters.refreshTimerLastTriggeredAt) {
-      updateDueToRefreshTimer = !_.isEqual(meta.refreshTimerLastTriggeredAt, filters.refreshTimerLastTriggeredAt);
+    if (refreshTimerAware && searchFilters.refreshTimerLastTriggeredAt) {
+      updateDueToRefreshTimer = !_.isEqual(meta.refreshTimerLastTriggeredAt, searchFilters.refreshTimerLastTriggeredAt);
     }
 
     let updateDueToFields = false;
     if (isFieldAware) {
-      updateDueToFields = !_.isEqual(meta.fieldNames, filters.fieldNames);
+      updateDueToFields = !_.isEqual(meta.fieldNames, searchFilters.fieldNames);
     }
 
     let updateDueToQuery = false;
     if (isQueryAware) {
-      updateDueToQuery = !_.isEqual(meta.query, filters.query);
+      updateDueToQuery = !_.isEqual(meta.query, searchFilters.query);
     }
 
+    let updateDueToPrecisionChange = false;
+    if (isGeohashPrecisionAware) {
+      updateDueToPrecisionChange = !_.isEqual(meta.geohashPrecision, searchFilters.geohashPrecision);
+    }
+
+    const updateDueToExtentChange = !this.updateDueToExtent(source, meta, searchFilters);
```

Comment:
Remove the `!` so this is `const updateDueToExtentChange = this.updateDueToExtent(source, meta, searchFilters);`. You added an extra negation since `updateDueToExtentChange ` is negated in the return statement